### PR TITLE
Refactor chat_app.py to use sseclient for improved SSE event handling

### PR DESCRIPTION
**Pull request by liucheng-ms**
# Description

This PR refactors the `chat_app.py` in our demo application to utilize the `sseclient` package for handling Server-Sent Events (SSE). The previous implementation relied on a custom `EventStream` class defined in `event_stream.py` to iterate over SSE events. By introducing the `sseclient` package, we can now delegate the SSE response to a more standardized and potentially more robust event list. 

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [X] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: docs/how-to-guides/enable-streaming-mode.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ If the response code is "424 Model Error", it means that the error is caused by
 
 ### Consume using Python
 
-We have created [a utility file](../media/how-to-guides/how-to-enable-streaming-mode/scripts/event_stream.py) as an example to demonstrate how to consume the server-sent event. A sample usage would like:
+A sample usage would like:
```
Comment (zzn2) [inline docs/how-to-guides/enable-streaming-mode.md:230]: We might need to say from where to import the SSEClient class and where to install the pypi package.

File: docs/media/how-to-guides/how-to-enable-streaming-mode/scripts/chat_app.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from azure.identity import AzureCliCredential
 import logging
 from azure.ai.ml import MLClient
-from event_stream import EventStream
+from sseclient import SSEClient
```
Comment (zzn2) [inline docs/media/how-to-guides/how-to-enable-streaming-mode/scripts/chat_app.py:9]: Maybe we need to add a requirements.txt to list up these external packages.

File: docs/how-to-guides/enable-streaming-mode.md [hunk 3]
```diff
@@ -236,8 +236,8 @@ try:
 
     content_type = response.headers.get('Content-Type')
     if "text/event-stream" in content_type:
-        event_stream = EventStream(response.iter_lines())
-        for event in event_stream:
+        client = SSEClient(response)
```
Comment (zzn2) [inline docs/how-to-guides/enable-streaming-mode.md:239]: We also need to remove the code of `EventStream` in this PR

**Review by zzn2 — APPROVED** [review]

File: docs/how-to-guides/enable-streaming-mode.md (hunk 1 above)
Comment (liucheng-ms) [reply docs/how-to-guides/enable-streaming-mode.md:235]: Thanks for the reminder, Zhidong! I've added the import and installation instructions to the snippet.

File: docs/how-to-guides/enable-streaming-mode.md (hunk 3 above)
Comment (liucheng-ms) [reply docs/how-to-guides/enable-streaming-mode.md:248]: Understood, Zhidong. I'll remove the EventStream code from the PR as suggested.

File: docs/media/how-to-guides/how-to-enable-streaming-mode/scripts/chat_app.py (hunk 2 above)
Comment (liucheng-ms) [reply docs/media/how-to-guides/how-to-enable-streaming-mode/scripts/chat_app.py:9]: Good idea, I'll add a requirements.txt for these dependencies.

**Review by zzn2 — APPROVED** [review]

**Review by chw-microsoft — APPROVED** [review]

**Pull request merged**